src/interceptors/authentication.interceptor.js

```diff
@@ -0,0 +1,37 @@
+const AuthenticationInterceptorModule = angular.module('tt2ps.interceptors.authentication', []);
+
+AuthenticationInterceptorModule.factory('AuthenticationInterceptor', AuthenticationInterceptor);
+
+AuthenticationInterceptor.$inject = ['SecurityContextService', 'AuthenticationService'];
+
+/**
+ * Intercepts HTTP requests to add authentication headers.
+ */
+function AuthenticationInterceptor(SecurityContextService, AuthenticationService) {
+    let securityEnabled = false;
+
+    SecurityContextService.onSecurityConfigurationChanged(
+        (securityConfiguration) => securityEnabled = securityConfiguration.enabled
+    );
+
+    return {
+        'request': function (config) {
+            const headers = config.headers || {};
+            // TODD maybe 'graphdb' have to be included
+            if (securityEnabled && config.url.includes('rest/')) {
```

Comment:
Do I understand correctly that this logic checks if an endpoint is secured? If so, so far we have the following endpoints 

```
POST /rest/chat/conversations
POST /rest/chat/conversations/explain/
GET /__health
GET /__gtg
GET /__about
GET /__trouble
```

According to https://github.com/statnett/Talk2PowerSystem_PM/issues/190 the new **insecure** endpoint exposing the auth settings must be named `/rest/authentication/config`. Hence, I think checking if an endpoint is secured with `config.url.includes('rest/')` is probably wrong.  For now, `config.url.includes('rest/chat')` would be better, but I'm not sure how the REST API will evolve in the future and what endpoints will be added, and if all of them will be under `rest/chat`